Production

Birdheezy opened the pull request.

Birdheezy committed:
Settings update with LED test function (#6)

* added git hub update settings.html

Added a git hub update button to settings.html

* Update blank.py

* Update blank.py

* changed button layout

* Create config_template.py

* Update settings.py

* deleted old files

* Update config_template.py

* Update config_template.py

* Update config_template.py

* Update config_template.py

* Update config_template.py

* Update config_template.py

* Update settings.py

* Update config_template.py

* Update config_template.py

* Update config_template.py

* config merge function final

* Update .gitignore

added airports.txt

* Update .gitignore

git ignore airprots.txt

* Update config.py

* Update settings.py

Updated Time stamp of last weather update

* Update update_cron.py

* Update .gitignore

* edited settings.py with def apply_updates and settings.js

* test

* updated print statement in metar.py

* Update .gitignore

* Update config.py

* Update metar.service

* Update settings.service

* Update weather.service

* Update settings.py

added backup creation to update route in settings.html/settings.py

* test

* Update settings.py

test

* Added scheduler..py and scheduler.service

* added scheduler.service

* test

* Update settings.html

re-ordered layout of settings.html

* Update settings.py

test

* Update config.py

test

* Update config.py

test

* small update

* Update scheduler.py

scheduler.py now dynamically updates lights on/off times

* Added "restart settings service"

Added a "restart settings service" button  to settings.html. Allows the user to fetch the git update then restart settings.py from the website.

* Update .gitignore

* Added to settings.html
- Color pickers for LED Colors
- Number of pixels setting

* Update config_template.py

updated default values

* Delete METARMap2.0.code-workspace

* Create README.md

* Added Legend

Added legend to metar.py and config..py

* Update README.md

* added legend toggle to settings.html

* Update settings.py

* updated settings.py to reload config settings
correctly when saving

* Update config_template.py

* Cleaned up "check for udpates" function
added rollback button

* Re added "restart settings service"...
not sure where it went...

* re-updated check for updates.

* Update README.md

* Update settings.css

* Added better logic to scheduler.py to turn lights
on/off based on current time

added toggle and setting in settings.html for
http/https.

* Update settings.html

uncommented "check for updates" button

* updated backup function again

* Update settings.html

#test to settings.html

* Update settings.html

removed #test from settings.html

* Update settings.py

* Update settings.py

* Update settings.py

changed owner of updated files to pi

* Update settings.py

* Update settings.py

update so config.py wouldn't be overwritten

* Update settings.py

* Update settings.py

* Update settings.py

* Update settings.py

* Update settings.py

* Updated service files and added PIXEL_PIN

Updated service files
added PIXEL_PIN setting to settings.html
had to convert PIXEL_PIN in metar.py and blank.py back to a string.

* removed files

Removed two unused files

* Update metar.py

* Update README.md

* Create setup.sh

* Update README.md

added setup.sh instructions

* Added: Weather Update toggle and interval to
settings.html, settings.py, scheduler.py,

updated setup.sh

* Update setup.sh

* Update setup.sh

* Update weather.json

* Update setup.sh

* Update setup.sh

* Update config.py

test value

* Update config.py

* Update config.py

* Update config.py

* Update config.py

* Update config.py

* Update config.py

* Update config.py

* Update config.py

* update settings.py

* added snowy color picker

* cleaned up settings.py inports

* only backup tracked files

* changed default lights off time

* config.py test9

* added tool tips to settings page html

* updated settings.py to preserve user settings and add new entries from repo

* test variable added to config.py to test
user values are preserved and new entries from repo are added

* updated settings.py to preserve user settings and add new entries from repo

* test for config.py

* Update config.py

* fixed eol in setup.sh

* updated setup.ssh ssl

* Update scheduler.py

* Update README.md

added youtube link

* added legend tooltip to settings.html

* Update README.md

Added to setup.sh instructions

* added service control panel

* removed unused buttons

* Added Service Control Panel for METAR, Settings, and Scheduler Services

- Added new UI section for service controls in settings.html
- Each service (METAR, Settings, Scheduler) has start/stop/restart buttons
- Added visual status indicators with colored dots
- Improved service status feedback
- Enhanced user interface with tooltips and status messages
- Added service control styling in settings.css

* undo

* update to settings.py

* log dropdown added

* try #2 for logging dropdown

* corrected why "show logs" was not working

* update button look

* added log buttons for metar, settings, and scheduler

* disabled line wrapping in logs

* changed service button locations

* formatting

* Update config.py

Updated default values

* Update settings.py

Updated message after successful update.

* Update settings.py

added wx reporting for HA integration

* update for HA integration

* updated toast messages for settings service restart

* removed unnecessary scheduler.py reload on save

* Update README.md

* Update scheduler.py

Stopped weather updates when metar.service is stopped

* cleaned up comments

Deleted comments

* modified metar.py and scheduler.py print statements

* removed unecessary print statements

* live weather update status in settings.html

* update to live weather update in settings.html

* removed console logging for live weather update in settings.html

* restore function try 1

* resore function try 2

* restore function try 2

* fixed removed service controls

* restore function try 3

* Update README.md

* added time zone setting

* updated time zone in settings.py

* updated tool tip icon and dropdown content

* updated tool tip icon and dropdown content

* updated look and feel of settings page
added timezone settings

* Settings.html update
updated color picker layout
added legend toggle with tooltip
Added new behavior to wx last updated text.
  - Text now turns red if the last update was more than 2x the refresh rate

* updated nav header for small screens

* major update to logging and scheduling

* update the the "weather last updated" status to be more descriptive
updated look of service section

* removed bootstrap.bundle.min.js

* updated gitignore

* restored to working settings.py file

* Update README.md

minor update

* Update README.md

added fix for setup.sh error

* change to printout of metar.py when lights are off

* Added Old weather LED animation
Added Wifi disconnection animation

* Made changes to update process to preserve user settings

* update settings.py

* Another attempt at an update function.

* Updated config.py and settings.py
new default values for config.py
updated config.py merge function

* service restart on update now

* test

* updated settings.py to restart services on update

* test

* test

* service restart order

* fixed settings service restart

* fixed pixel pin tooltip

* fix for color settings and saving config

* added LED color order to settings

* updated git commands for update function

* Delete settings.html

* cleaned up config.py

* Added live weather map view and saved view functionality
added local IP address display to settings page
updated file permissions set after update

* added kiosk mode

* deleted .vscode

* kiosk mode

* major revamp of kiosk page.
added map dispaly
cleaned up look of page
added weather status to kiosk page

* small update to kiosk mode

* Update to weather.py to use absolute path
update to kiosk.html and settings.html to live update weather status dot and time

* Updated Kiosk Map to be live updated and moved map to top of page

* corrected color order settings for LEDs.

* fixed empty backup directory

* better logging for backup

* added wx update to file update process

* Update README.md

added download and execute command to setup.sh

* Update README.md

update to connecting to pi instructions

* updated readme file for connecting to pi

* updated setup.sh

* small update to settings.py

* updated look of settings page with better styling
updated map.js to use new styling
updated map legond dots to be dynamic

* Update .gitignore

* fixed <section> tags

* Added LED testing functionality and UI improvements

- Added led_test.py script for LED testing capabilities
- Created systemd service file for LED testing
- Enhanced settings page with LED testing section
- Improved CSS styling for settings page
- Added automated and manual LED testing options
- Updated setup script to handle new LED test service
- Minor fixes and improvements to settings handling

* re order some items in settings.html

* Small changes to settings.html

Small changes to settings.html

* Updated settings functionality with improved UI and configuration options

- Enhanced settings page layout and styling
- Added new configuration options in settings.py
- Updated JavaScript functionality for settings management
- Improved settings template structure

* Updated configuration settings in settings.py:
- Added time variables for light control (on/off times, brightness scheduling)
- Added color definitions for different flight conditions and status indicators
- Added animation settings for wind, snow, and lightning effects
- Added configuration flags for various display features
- Added map center coordinates and zoom level settings

* feat: Major updates to settings and scheduler functionality

- Added dynamic weather update scheduling based on config
- Improved error handling and logging in scheduler
- Enhanced settings UI with new styling and controls
- Added map.js functionality for better visualization
- Updated setup script for improved installation
- Implemented config file monitoring for live updates
- Fixed led test functionality

* Fixed scheduler not restarting metar service on config.py changes
Fixed "save" button restarting metar service unnecessarily
big update to led test section
Improved settings page layout and styling
Added validation for config file changes
Added weather update on lights on
Fixed nav bar layout
fix "wx last updated" status dot and text not being live

* Major UI/UX improvements and code refactoring

- Extracted map component into separate map.html template for reuse
- Enhanced map styling with improved popups and legend
- Added weather status indicator with last update time
- Improved settings page layout and styling
- Added map view saving functionality
- Updated JavaScript code organization and error handling
- Refined LED testing interface
    - Added RGB/BRG pixel order selection
    - Added single or multiple LED testing options
- Added configuration options for map center and zoom
- General code cleanup and documentation improvements

Birdheezy committed:
Update led_test.py

Fixed all LEDs lighting up when only a certain range was selected.

Birdheezy committed:
- fixed wx last updated status dot to updated in real time
 - fixed airport dots on map to update in real time
 - added a button to start the METAR service in LED test section
 - minor layout changes to settings page

Birdheezy committed:
Merge branch 'production' of https://github.com/Birdheezy/METARMap2.0 into production

Birdheezy committed:
big update

updated production branch to match main branch due to aviationweather.gov changing API

Birdheezy committed:
fix for aviationweather.gov API changes

Birdheezy committed:
updated metar.py for "none" in logs

Birdheezy committed:
changed log formatting for "none" in logs

Birdheezy committed:
fix metar parsing and clean up settings,html

Added METAR raw parsing to fix fltcat missing on airports that don't report certain criteria.
update to settings.html.
added new snowy animation
removed "raw metar" head from the map in settings.html pop up bubble

Trombs commented:
README instructions are resulting in an error for me. Unable to get the script to work.

pi@raspberrypi:~ $ metar metar.py
Traceback (most recent call last):
  File "/home/pi/metar.py", line 6, in <module>
    import neopixel
  File "/home/pi/metar/lib/python3.13/site-packages/neopixel.py", line 20, in <module>
    from neopixel_write import neopixel_write
  File "/home/pi/metar/lib/python3.13/site-packages/neopixel_write.py", line 22, in <module>
    from adafruit_blinka.microcontroller.bcm283x import neopixel as _neopixel
  File "/home/pi/metar/lib/python3.13/site-packages/adafruit_blinka/microcontroller/bcm283x/neopixel.py", line 7, in <module>
    import _rpi_ws281x as ws
ModuleNotFoundError: No module named '_rpi_ws281x'

Birdheezy commented:
It looks like the neopixel library was not installed. Did you run the setup.sh script? With sudo?
Sudo bash setup.sh
> README instructions are resulting in an error for me. Unable to get the script to work.
> 
> pi@raspberrypi:~ $ metar metar.py
> Traceback (most recent call last):
>   File "/home/pi/metar.py", line 6, in <module>
>     import neopixel
>   File "/home/pi/metar/lib/python3.13/site-packages/neopixel.py", line 20, in <module>
>     from neopixel_write import neopixel_write
>   File "/home/pi/metar/lib/python3.13/site-packages/neopixel_write.py", line 22, in <module>
>     from adafruit_blinka.microcontroller.bcm283x import neopixel as _neopixel
>   File "/home/pi/metar/lib/python3.13/site-packages/adafruit_blinka/microcontroller/bcm283x/neopixel.py", line 7, in <module>
>     import _rpi_ws281x as ws
> ModuleNotFoundError: No module named '_rpi_ws281x'

Birdheezy committed:
Increase maximum value limit from 50 to 500

this value would limit the number of LED's you could use in a map. upped to 500

Birdheezy committed:
Update weather.py

update for headers for aviationweather.gov API update

Birdheezy committed:
Merge branch 'production' of https://github.com/Birdheezy/METARMap2.0 into production
